From 719d4d215eba086e206f7743f541b90f55c41f29 Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:31:14 -0400
Subject: [scsi] hpsa: get physical device handles for io accel mode 2 as well as mode 1

Message-id: <1410186734-19986-24-git-send-email-thenzl@redhat.com>
Patchwork-id: 92037
O-Subject: [RHEL6.6 e-stor PATCH 23/83] hpsa: get physical device handles for io accel mode 2 as well as mode 1
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

Signed-off-by: Mike MIller <michael.miller@canonical.com>
Signed-off-by: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit 317d4adfd37729452c840821c71d1162d5d63f39)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index 3a58382..a3a6a3c 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -2144,7 +2144,8 @@ static int hpsa_gather_lun_info(struct ctlr_info *h,
  *physical_mode = 0;
 
  /* For I/O accelerator mode we need to read physical device handles */
- if (h->transMethod & CFGTBL_Trans_io_accel1) {
+ if (h->transMethod & CFGTBL_Trans_io_accel1 ||
+  h->transMethod & CFGTBL_Trans_io_accel2) {
   *physical_mode = HPSA_REPORT_PHYS_EXTENDED;
   physical_entry_size = 24;
  }
-- 
1.7.1